### PR TITLE
Document example workflows with upload and also test them

### PR DESCRIPTION
**Pull request by sverhoeven**
Refs #79

## Description

* Shows how to get the HTML files out of a workflow.
* Make the workflows shown in README the same as those in `.github/workflows/usage.yml`.
* The `.github/workflows/usage.yml` workflow no longer runs on each PR as `.github/workflows/testing.yml` tests the action much better. It can still be triggered manually see https://github.com/tortellini-tools/action/actions/runs/927859407 and will be run when main branch is changed.

* * *

## Before you get started

- [x] I read and I followed [contributing guidelines](https://github.com/tortellini-tools/action/blob/main/CONTRIBUTING.md)
- [x] [☝ Create an issue](https://github.com/NLeSC/licenseguard/issues/new/choose) to discuss what you are going to do

## Checklist

This section should be filled in by the creator of this pull request to make sure the pull request is ready to review.

- [ ] This pull request has a descriptive title
- [ ] Code is written according to the code quality guidelines
- [ ] Documentation if applicable is available
- [ ] Tests run successfully
- [ ] All checks below this pull request were successful

### CODE REVIEW

File: .github/workflows/usage.yml [hunk 1]
```diff
@@ -11,25 +10,26 @@ on: # run any PRs and main branch changes
   workflow_dispatch:
 
 jobs:
-
-  tortellini:
+  current-repository:
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v2
```
Comment (jspaaks) [inline .github/workflows/usage.yml]: unrelated to this PR but just a thought: maybe we should make checking out the code part of tortellini. Something like, 
```
if (repositories==='') {
    find out which remote to use, or use GITHUB_* env vars
    git clone that remote
    proceed as normal
}
```

File: .github/workflows/usage.yml [hunk 2]
```diff
@@ -11,25 +10,26 @@ on: # run any PRs and main branch changes
...
+      - uses: tortellini-tools/action@main
+      - uses: actions/upload-artifact@v2
+        with:
+          name: tortellini-result
+          path: .tortellini/out
+
+  multiple-repositories:
+    runs-on: ubuntu-latest
+    steps:
       - run: |
```
Comment (jspaaks) [inline .github/workflows/usage.yml]: Consider adding a name to this `run` to explain why we need this

File: README.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-  <a href="https://github.com/tortellini-tools/action/actions"><img alt="linting-action status" src="https://github.com/tortellini-tools/action/workflows/linting/badge.svg"></a>  
-  <a href="https://github.com/tortellini-tools/action/actions"><img alt="tortellini-action status" src="https://github.com/tortellini-tools/action/workflows/tortellini/badge.svg"></a>
+  <a href="https://github.com/tortellini-tools/action/actions"><img alt="linting-action status" src="https://github.com/tortellini-tools/action/workflows/linting/badge.svg"></a>
+  <a href="https://github.com/tortellini-tools/action/actions/workflows/usage.yml"><img alt="tortellini-action status" src="https://github.com/tortellini-tools/action/actions/workflows/usage.yml/badge.svg"></a>
 </p>
 
 # Tortellini GitHub Action
 
-This action checks dependency licence issues using [ort](https://github.com/oss-review-toolkit/ort).
+This action checks dependency license issues using [ort](https://github.com/oss-review-toolkit/ort).
```
Comment (jspaaks) [inline README.md:10]: that's right y'all!

File: README.md [hunk 4]
```diff
@@ -21,44 +21,55 @@ This action checks dependency licence issues using [ort](https://github.com/oss-
...
+            - uses: actions/checkout@v2
+            - uses: tortellini-tools/action@main
+            - uses: actions/upload-artifact@v2
+              with:
+                  name: tortellini-result
+                  path: .tortellini/out
 ```
 
+Tortellini action will generate `.tortellini/out/scan-report-web-app.html` file.
+The HTML file can be downloaded from the [workflow page](https://github.com/actions/upload-artifact#where-does-the-upload-go) by using the GitHubs upload-artifact action.
```
Comment (jspaaks) [inline README.md]: This sentence is a bit confusing to me. I would not expect a pointer to "the workflow page" to point to the documentation of the action.

Maybe something like:

The HTML file can be downloaded from the workflow summary page (see the documentation of the [`upload-artifact` Github Action](https://github.com/actions/upload-artifact#where-does-the-upload-go).

File: README.md [hunk 5]
```diff
@@ -21,44 +21,55 @@ This action checks dependency licence issues using [ort](https://github.com/oss-
...
-        echo 'https://github.com/fair-software/howfairis' >> urls.txt
-    - uses: tortellini-tools/action@main
-      with:
-        repositories: 'urls.txt'
-        rules: https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/rules.kts
-        classifications: https://github.com/NLeSC/tortellini-on-rsd/raw/main/config/license-classifications.yml
+    tortellini:
+        runs-on: ubuntu-latest
+        steps:
+            - run: |
```
Comment (jspaaks) [inline README.md]: use `name` to describe why you need this step

File: README.md [hunk 6]
```diff
@@ -21,44 +21,55 @@ This action checks dependency licence issues using [ort](https://github.com/oss-
...
+              with:
+                  repositories: urls.txt
+            - uses: actions/upload-artifact@v2
+              with:
+                  name: tortellini-results
+                  path: .tortellini/out
 ```
 
+Tortellini action will generate `.tortellini/out/index.html` file and `.tortellini/out/<GitHub user or organization>/<GitHub repository>/scan-report-web-app.html` files.
+The HTML files can be downloaded from [your workflow page](https://github.com/actions/upload-artifact#where-does-the-upload-go) by using GitHubs upload-artifact action.
```
Comment (jspaaks) [inline README.md]: Same comment as above

File: README.md [hunk 7]
```diff
@@ -21,44 +21,55 @@ This action checks dependency licence issues using [ort](https://github.com/oss-
...
+
+To view the HTML files directly you could upload the `.tortellini/out` folder to a web server like an AWS S3 bucket.
```
Comment (jspaaks) [inline README.md]: Consider covering just viewing separately from hosting

**Review by jspaaks — COMMENTED** [review]
Thanks for the PR. I left a couple of comments. Apart from those, should we consider offering the two usage cases as separate workflow files? something like `usage-current-repository.yml` and `usage-multiple-repositories.yml`? (we can leave that for another day, just wanted to get your opinion)

File: .github/workflows/usage.yml (hunk 1 above)
Comment (sverhoeven) [reply .github/workflows/usage.yml]: Created https://github.com/tortellini-tools/action/issues/139

File: README.md (hunk 3 above)
Comment (sverhoeven) [reply README.md:10]: Yippee-ki-yay

**Comment by sverhoeven** [comment]
> Thanks for the PR. I left a couple of comments. Apart from those, should we consider offering the two usage cases as separate workflow files? something like `usage-current-repository.yml` and `usage-multiple-repositories.yml`? (we can leave that for another day, just wanted to get your opinion)

Thanks for the comments, I addressed all of them.
I like you proposal. We are showing complete workflows in the README having them separated in `.github/workflows` make sense. It will make it easier to keep them the same (README vs .github/workflows).

File: README.md (hunk 5 above)
Comment (sverhoeven) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (sverhoeven) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (sverhoeven) [reply README.md]: done, dropped hosting in favor of viewing

**Review by jspaaks — APPROVED** [review]
Good stuff, thank you Stefan!

**Pull request merged**